server/Helpers/search.js

```diff
@@ -1,46 +1,78 @@
 import axios from "axios";
 import qs from "qs";
 
-const ARCHIVE_URL = "https://archive.org/advancedsearch.php";
-
-const searchHelper = async ({
-  query,
-  page,
-  minRating,
-  maxRating,
-  startYear,
-  endYear,
-  collections
-}) => {
+// const ARCHIVE_URL = "https://archive.org/advancedsearch.php";
+
+const YTS_URL = "https://yts.lt/api/v2/list_movies.json";
+const POPCORN_URL = "https://tv-v2.api-fetch.website/shows";
+
+const searchMovies = async ({ query, page, minRating, year, collections }) => {
   try {
-    const formattedDate =
-      startYear && endYear
-        ? ` date:[${startYear}-01-01 TO ${endYear}-12-31]`
-        : "";
-    const formattedRating =
-      minRating && maxRating
-        ? ` avg_rating:[${minRating} TO ${maxRating}]`
-        : " ";
-    const formattedCollections = collections
-      ? ` collection:(${collections.join(" OR ")})`
-      : "";
-    const formattedQuery = query ? ` ${query}` : "";
-
-    const initialQuery = `mediatype:"movies" format:"Archive BitTorrent"${formattedCollections}${formattedDate}${formattedRating}${formattedQuery}`;
-    const archiveSearchQuery = {
-      q: initialQuery,
-      sort: ["downloads desc", "date desc", "titleSorter asc"],
-      rows: "10",
+    const queryParams = qs.stringify({
+      limit: 12,
+      sort_by: "download_count",
+      minimum_rating: minRating * 2 || 0,
       page,
-      output: "json"
+      query_term: `${query || ""} ${year || ""}`,
+      genre: collections
+    });
+
+    const { data } = await axios.get(`${YTS_URL}?${queryParams}`);
+
+    const parsedMovies =
+      (data &&
```

Comment:
```suggestion
      (data &&
        data.data &&
        data.data.movies && 
```

When there's no result, `data` exist but `data.data.movies` is `undefined`. This cause an error `Cannot read property 'map' of undefined` that is catch by the try/catch.

Then the api respond with a 404 status and then when you make a search on movies (not show) that has no result you hand up being redirected to `localhost:8080/error` (because of 404) (and you don't get the chance to see the cute pop corn ! What a shame)